libevmasm/Assembly.cpp

```diff
@@ -227,16 +239,19 @@ AssemblyItem Assembly::createAssemblyItemFromJSON(Json const& _json, std::vector
 		else if (name == "PUSH [tag]")
 		{
 			requireValueDefinedForInstruction(name, value);
+			requireValueInRange(name, value, 0, std::numeric_limits<unsigned>::max());
 			result = {AssemblyItemType::PushTag, updateUsedTags(u256(value))};
 		}
 		else if (name == "PUSH [$]")
 		{
 			requireValueDefinedForInstruction(name, value);
+			requireValueInRange(name, "0x" + value, 0, std::numeric_limits<size_t>::max());
```

Comment:
We really shouldn't be using platform-dependent types here, like `unsigned` or `size_t`. This means that validation ranges are platform dependent.